lib/nxdk/configsector.c

```diff
@@ -147,3 +148,66 @@ bool nxLoadNetworkConfig(nxdk_network_config_sector_t *config)
 
     return true;
 }
+
+bool nxSaveNetworkConfigSector(HANDLE hPartition, UINT sectorIndex, const BYTE *buffer, UINT size)
+{
+    NXDK_CONFIG_SECTOR configSector;
+
+    _Static_assert(sizeof(configSector) == 512, "NXDK_CONFIG_SECTOR size mismatch");
```

Comment:
To me, this feels like implicit documentation that config sectors (`NXDK_CONFIG_SECTOR`) are 512 bytes.

Therefore, I think I'd prefer if this was right below the struct definition of `NXDK_CONFIG_SECTOR` and not whereever `NXDK_CONFIG_SECTOR` is used, unless that's important for the task.

In this case, it might make sense to check against `NXDK_HDD_SECTOR_SIZE`? The function implies that this expects both to be the same value.

If you keep the assert here, it should be moved further down in my opinion (to whereever the assumption actually matters).

Reply:
It's a safeguard against accidentally changing the size when the struct gets modified in the future when the meaning or type of more fields is figured out. I've de-duplicated and moved up the assert, and also added one for `nxdk_network_config_sector_t` to protect it, too.